### ISSUE TITLE
Composer 2 RC1 warning: plugin not handling the returned Promise from the downloader

### ISSUE THREAD

**Issue by LordSimal** [cakephp/plugin-installer#47]
I just updated my Composer to the current Composer 2 RC1 and tried to re-install all my PHP modules with
`composer install`

This now causes the following warning messages for multiple composer modules like this

```
Failed to extract dereuromark/cakephp-tools: (9) unzip -qq  ' <my-path>/vendor/composer/tmp-a05977d85b922d82afd80c9af3c81a17' -d ' <my-path>/vendor/composer/91dc78cf'

unzip:  cannot find or open <my-path>/vendor/composer/tmp-a05977d85b922d82afd80c9af3c81a17.

    This most likely is due to a custom installer plugin not handling the returned Promise from the downloader
    See https://github.com/composer/installers/commit/5006d0c28730ade233a8f42ec31ac68fb1c5c9bb for an example fix
```

The link from the message is: https://github.com/composer/installers/commit/5006d0c28730ade233a8f42ec31ac68fb1c5c9bb

When I remove the `cakephp/plugin-installer` module from my composer.json and reinstall all modules this warning is gone.

Therefore I think something needs to be adapted for Composer 2 to work correctly.

I am currently on CakePHP 4.1.4 but I think this has nothing to do with the used CakePHP version.

**Comment by ADmad** [comment]
What's the version of `plugin-installer` you are using and what does your app's `composer.json` look like?

**Comment by LordSimal** [comment]
I am using cakephp/plugin-installer version 1.2

My composer.json
```
{
  "name": "cakephp/app",
  "description": "CakePHP skeleton app",
  "homepage": "https://cakephp.org",
  "type": "project",
  "license": "MIT",
  "require": {
    "php": ">=7.2",
    "ext-mysqli": "*",
    "ext-curl": "*",
    "ext-ftp": "*",
    "ext-intl": "*",
    "ext-json": "*",
    "bitbucket/client": "^3.1",
    "cakephp/cakephp": "^4.1",
    "cakephp/migrations": "^3.0.0",
    "cakephp/plugin-installer": "^1.2",
    "dereuromark/cakephp-queue": "^5.0",
    "dereuromark/cakephp-tools": "^2.1",
    "dereuromark/cakephp-setup": "^2.0",
    "easybill/php-sdk": "^2.0",
    "gamez/mite": "^1.0",
    "guzzlehttp/guzzle": "^6.5",
    "http-interop/http-factory-guzzle": "^1.0",
    "jeroendesloovere/vcard": "^1.7",
    "lesstif/php-jira-rest-client": "^2.4",
    "mlocati/ip-lib": "^1.12",
    "mobiledetect/mobiledetectlib": "^2.8",
    "php-http/guzzle6-adapter": "^2.0.1",
    "phpoffice/phpspreadsheet": "^1.9",
    "phpseclib/phpseclib": "^2.0",
    "solarium/solarium": "^6.0",
    "stevenmaguire/oauth2-bitbucket": "^3.0",
    "symfony/event-dispatcher": "^5.1"
  },
  "require-dev": {
    "cakephp/bake": "^2.0",
    "cakephp/cakephp-codesniffer": "^4.1",
    "cakephp/debug_kit": "^4.0",
    "dereuromark/cakephp-ide-helper": "^1.5",
    "dereuromark/cakephp-test-helper": "^1.0",
    "josegonzalez/dotenv": "^3.2",
    "pakacuda/cakephp-fixture-factories": "^0.2.4",
    "phpunit/phpunit": "^8.5",
    "psy/psysh": "@stable",
    "phan/phan": "^2.4"
  },
  "suggest": {
    "markstory/asset_compress": "An asset compression plugin which provides file concatenation and a flexible filter system for preprocessing and minification.",
    "dereuromark/cakephp-ide-helper": "After baking your code, this keeps your annotations in sync with the code evolving from there on for maximum IDE and PHPStan compatibility."
  },
  "autoload": {
    "psr-4": {
      "App\\": "src/"
    }
  },
  "autoload-dev": {
    "psr-4": {
      "App\\Test\\": "tests/",
      "Cake\\Test\\": "vendor/cakephp/cakephp/tests/"
    }
  },
  "scripts": {
    "post-install-cmd": "App\\Console\\Installer::postInstall",
    "post-create-project-cmd": "App\\Console\\Installer::postInstall",
    "post-autoload-dump": "Cake\\Composer\\Installer\\PluginInstaller::postAutoloadDump",
    "check": [
      "@test",
      "@cs-check"
    ],
    "cs-check": "phpcs --colors -p --standard=vendor/cakephp/cakephp-codesniffer/CakePHP src/ tests/",
    "cs-fix": "phpcbf --colors --standard=vendor/cakephp/cakephp-codesniffer/CakePHP src/ tests/",
    "stan": "phpstan analyse src/",
    "stan-setup": "cp composer.json composer.backup && composer require --dev phpstan/phpstan-shim:^0.11 && mv composer.backup composer.json",
  },
  "prefer-stable": true,
  "config": {
    "sort-packages": true,
    "process-timeout": 0
  }
}
```

**Comment by ADmad** [comment]
Fixed in 1.3.0 release.
